scalableminds/webknossos
# Add option to calculate segment index for an existing volume tracing

This PR adds a new button to the volume layer menu options that allows the user to calculate the segment index for the volume annotation.

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
I recommend testing locally as the backend currently sets `hasSegmentIndex` to true as default for new annotations.
- In file `AnnotationService.scala` change `hasSegmentIndex = Some(fallbackLayer.isEmpty)` to `hasSegmentIndex = Some(false)` (in line 166)
- Create a new annotation with a volume annotation without a fallback layer
- Use the new option in the dropdown menu of the volume layer and click the "Add segment index to this layer" button.
- A modal should open up informing about this feature
- Click the "Calculate Segment Index" button
- The modal should not close, the backend should do a short calculation and afterwards the whole page should reload.
- The option to calculate the segment index should no longer exist in a real world scenario as the index has already be calculated (but might still be possible during testing due to the changes in `AnnotationService.scala`) 


### Issues:
- fixes [#7144](https://github.com/scalableminds/webknossos/issues/7144)

------
(Please delete unneeded items, merge only when none are left open)
